Automattic/wp-calypso
# Site Settings: Display warning dialog in `DeleteSite` when deleting a site with active subscriptions

This PR:

- Moves the warning dialog from `DeleteSiteOptions` into a new component.
- Removes logic to fetch site purchases from `DeleteSiteOptions`.
- Adds a data component, `SitePurchasesData`, and uses it to provide site purchases data to `DeleteSite` and `DeleteSiteOptions`.
- Updates the site settings controller to use JSX.

Fixes #3117.

<img width="559" alt="screen shot 2016-03-30 at 5 47 45 pm" src="https://cloud.githubusercontent.com/assets/1130674/14162476/1189d556-f6a1-11e5-8600-cc9ed764bcfa.png">

**Testing**
*Sites without active subscriptions*
- Visit `/settings/general/:site` for a site without active subscriptions.
- Assert that the "Delete Site" link at the bottom of the page appears after purchases have loaded.
- Click the "Delete Site" link.
- Assert that you are taken to `/settings/delete-site/:site`.

*Sites with active subscriptions*
- Visit `/settings/general/:site` for a site with active subscriptions.
- Assert that the "Delete Site" link at the bottom of the page appears after purchases have loaded.
- Click the "Delete Site" link.
- Assert that the dialog above appears.
- Visit `/settings/delete-site/:site` for a site with active subscriptions in a new session.
- Assert that the "Delete Site" button at the bottom is disabled until purchases have loaded.
- Click "Delete Site" once it is enabled.
- Assert that the dialog above appears.

**Note:** Actually deleting a site is buggy, right now. I opened [an issue](https://github.com/Automattic/wp-calypso/issues/4444) for this.
